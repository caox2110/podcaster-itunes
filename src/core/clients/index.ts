export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
