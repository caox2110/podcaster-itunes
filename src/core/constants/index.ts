import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
