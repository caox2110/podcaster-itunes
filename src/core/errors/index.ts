export { default as ExceptionHandler } from './Exceptions.handler';
