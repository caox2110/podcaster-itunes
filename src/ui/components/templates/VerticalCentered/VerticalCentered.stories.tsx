import type { ComponentStory, ComponentMeta } from '@storybook/react';
import { withPerformance } from 'storybook-addon-performance';

import VerticalCentered from '.';

const Template: ComponentStory<typeof VerticalCentered> = (properties) => (
  <VerticalCentered {...properties} />
);

const VerticalCenteredWithTitle = Template.bind({});
VerticalCenteredWithTitle.args = {
  children: <h1>Testing Text Centered</h1>,
};

export { VerticalCenteredWithTitle };
export default {
  title: 'UI/Templates/VerticalCentered',
  component: VerticalCentered,
  parameters: {
    layout: 'fullscreen',
  },
  decorators: [withPerformance],
} as ComponentMeta<typeof VerticalCentered>;
