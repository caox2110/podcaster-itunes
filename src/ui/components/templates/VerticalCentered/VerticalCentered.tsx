import { ReactNode } from 'react';
import type { BoxProps } from '@mui/material/Box';
import type { Theme } from '@mui/material/styles';

import { MuiBox, MuiGrid } from '../..';

/**
 * Type of VerticalCentered component itÂ´s can receive the BoxProps too
 *
 * @typedef {VerticalCenteredProperties}
 */
type VerticalCenteredProperties = {
  /**
   * Child component of VerticalCentered component
   *
   * @type {?ReactNode}
   */
  children?: ReactNode;
} & BoxProps;

/**
 * Default values for VerticalCentered component props
 *
 * @type {{ children: any; className: string; }}
 */
const defaultProps = {
  children: null,
  className: '',
};

VerticalCentered.defaultProps = defaultProps;

/**
 * Component to center the childs in the screen
 *
 * @param {VerticalCenteredProperties} { children, className, ...rest }
 * @returns
 */
function VerticalCentered({ children, ...rest }: VerticalCenteredProperties) {
  return (
    <MuiBox
      component='div'
      sx={{
        flexGrow: 1,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        minHeight: '100vh',
        textAlign: 'center',
      }}
      {...rest}
    >
      <MuiGrid
        container
        sx={{
          justifyContent: 'center',
          padding: (theme: Theme) => theme.spacing(3),
        }}
      >
        <MuiGrid item lg={6} md={8} sm={10} xs={12}>
          {children}
        </MuiGrid>
      </MuiGrid>
    </MuiBox>
  );
}

export default VerticalCentered;
