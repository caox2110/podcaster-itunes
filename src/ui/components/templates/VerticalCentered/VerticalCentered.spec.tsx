import { render, screen } from '@/core/test';
import type { RenderResult } from '@/core/test';

import VerticalCentered from '.';

function renderApp(): RenderResult {
  const utils = render(
    <VerticalCentered>
      <h1>Testing text</h1>
    </VerticalCentered>,
  );
  return {
    ...utils,
  };
}

describe('VerticalCentered', () => {
  describe('Rendering', () => {
    it('Should render without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('Should render a children without throwing any errors', () => {
      renderApp();

      expect(screen.getByText(/testing text/i)).toBeInTheDocument();
    });
  });
});
