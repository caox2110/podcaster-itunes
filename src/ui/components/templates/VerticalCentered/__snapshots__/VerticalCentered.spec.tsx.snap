// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalCentered Rendering Should render without throwing any errors 1`] = `
<div>
  <div
    class="MuiBox-root css-fulu7r"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1lrzoir-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-8 MuiGrid-grid-lg-6 css-1cotp1z-MuiGrid-root"
      >
        <h1>
          Testing text
        </h1>
      </div>
    </div>
  </div>
</div>
`;
